mm2src/coins/eth.rs

```diff
@@ -929,34 +929,52 @@ impl SwapOps for EthCoin {
                     .compat()
                     .await
             );
+
             if status == PAYMENT_STATE_UNINITIALIZED.into() {
                 return Ok(None);
             };
-            let events = try_s!(
-                selfi
-                    .payment_sent_events(swap_contract_address, from_block)
-                    .compat()
-                    .await
-            );
 
-            let found = events.iter().find(|event| &event.data.0[..32] == id.as_slice());
+            let mut current_block = try_s!(selfi.current_block().compat().await);
+            if current_block < from_block {
```

Comment:
I'm confused a bit :D
When can it be possible? If I understand correctly, `from_block` is set to a current block number when a swap starts.

Reply:
A remote node might be out of sync or someone can occasionally use a node from the wrong network.